src/index.ts

```diff
@@ -707,20 +723,29 @@ class Envoy {
     const app_state = this.app_states[installed_app_id]
 
     let activation_interval
+    const when_this_interval = Date.now().toLocaleString()
 
     // let the record show, we're not happy with this code.
     try {
       await new Promise<void>((resolve, reject) => {
         app_state.desired_activation_state = 'activated'
+        log.normal('setting desired state to activated. actual: %s', this.app_states[installed_app_id].desired_activation_state)
+        app_state.desired_activation_state_changed_at = Date.now()
+
 
         const tryToActivate = async () => {
+          log.normal("trying to activate. desired_activation_state: %s", app_state.desired_activation_state)
+          log.normal("this interval initiated at: %s", when_this_interval)
+          log.normal("app_state: %s", inspect(app_state))
+
+
           if (app_state.desired_activation_state === 'deactivated') {
```

Comment:
Looks like this condition will never be true since this value is being set immediately above it, on line 732.  I think we can still get the desired outcome though, if instead we focus on the `'deactivating'` state.
```suggestion
          if (app_state.activation_state === 'deactivating') {
```

Reply:
Again, this is in an interval, this code runs 5000/10,000 or more seconds after the code above it that sets the desired state to `deactivated`